rero_ils/modules/documents/serializers/dc.py

```diff
@@ -0,0 +1,175 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2022 RERO
+# Copyright (C) 2022 UCLouvain
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""DublinCore Document serialization."""
+
+from dcxml import simpledc
+from flask import request
+from invenio_records_rest.serializers.dc import \
+    DublinCoreSerializer as BaseDublinCoreSerializer
+from lxml import etree
+from lxml.builder import ElementMaker
+
+from rero_ils.modules.documents.api import Document
+from rero_ils.modules.documents.dojson.contrib.jsontodc import dublincore
+from rero_ils.modules.documents.utils import create_contributions
+
+DEFAULT_LANGUAGE = 'en'
+
+
+class DublinCoreSerializer(BaseDublinCoreSerializer):
+    """Dublin Core serializer for records.
+
+    Note: This serializer is not suitable for serializing large number of
+    records.
+    """
+
+    # Default namespace mapping.
+    namespace = {
+        'dc': 'http://purl.org/dc/elements/1.1/',
+        'xml': 'xml',
+    }
+    # Default container element attributes.
+    # TODO: save local dc schema
+    container_attribs = {
+        '{http://www.w3.org/2001/XMLSchema-instance}schemaLocation':
+        'https://www.loc.gov/standards/sru '
+        'https://www.loc.gov/standards/sru/recordSchemas/dc-schema.xsd',
+    }
+    # Default container element.
+    container_element = 'record'
+
+    def transform_record(self, pid, record, links_factory=None,
+                         language=DEFAULT_LANGUAGE, **kwargs):
+        """Transform record into an intermediate representation."""
+        record = record.replace_refs()
+        contributions = create_contributions(
+            record.get('contribution', [])
+        )
+        if contributions:
+            record['contribution'] = contributions
+        record = dublincore.do(record, language=language)
+        return record
+
+    def transform_search_hit(self, pid, record, links_factory=None,
+                             language=DEFAULT_LANGUAGE, **kwargs):
+        """Transform search result hit into an intermediate representation."""
+        record = Document.get_record_by_pid(pid)
+        return self.transform_record(
+            pid=pid,
+            record=record,
+            links_factory=links_factory,
+            language=language,
+            **kwargs
+        )
+
+    # Needed if we use it for documents serialization !
```

Comment:
Can we remove this?

Reply:
yes normally, but we keep it in case we want to do marcxml for something other than the SRU

Reply:
ok